services/src/floodlight-service/floodlight-modules/src/main/java/org/openkilda/floodlight/switchmanager/SwitchManager.java

```diff
@@ -645,27 +667,75 @@ public long installOneSwitchFlow(DatapathId dpid, String flowId, Long cookie, in
                 addLldpIngressFlow(flows, sw);
             }
         }
-        ArrayList<OFAction> actionListBroadcastRule = prepareActionListForBroadcastRule(sw);
-        OFInstructionMeter meterBroadcastRule = buildMeterInstructionForBroadcastRule(sw, actionListBroadcastRule);
-        Optional.ofNullable(buildVerificationRule(sw, true, ofFactory, VERIFICATION_BROADCAST_RULE_COOKIE,
-                meterBroadcastRule, actionListBroadcastRule)).ifPresent(flows::add);
+        return flows;
+    }
 
-        ArrayList<OFAction> actionListUnicastRule = prepareActionListForUnicastRule(sw);
-        OFInstructionMeter meterUnicastRule = buildMeterInstructionForUnicastRule(sw, actionListUnicastRule);
-        Optional.ofNullable(buildVerificationRule(sw, false, ofFactory, VERIFICATION_UNICAST_RULE_COOKIE,
-                meterUnicastRule, actionListUnicastRule)).ifPresent(flows::add);
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public List<MeterEntry> getExpectedDefaultMeters(DatapathId dpid, boolean multiTable, boolean switchLldp)
+            throws SwitchOperationException {
+        IOFSwitch sw = lookupSwitch(dpid);
+        boolean packetsFlag = featureDetectorService.detectSwitch(sw).contains(SwitchFeature.PKTPS_FLAG);
+
+        List<SwitchFlowGenerator> defaultFlowGenerators = new ArrayList<>();
+        defaultFlowGenerators.add(switchFlowFactory.getDropFlowGenerator(DROP_RULE_COOKIE, MINIMAL_POSITIVE_PRIORITY,
+                INPUT_TABLE_ID));
+        defaultFlowGenerators.add(switchFlowFactory.getVerificationFlow(true));
+        defaultFlowGenerators.add(switchFlowFactory.getVerificationFlow(false));
+        defaultFlowGenerators.add(switchFlowFactory.getDropLoopFlowGenerator());
+        defaultFlowGenerators.add(switchFlowFactory.getBfdCatchFlowGenerator());
+        defaultFlowGenerators.add(switchFlowFactory.getRoundTripLatencyFlowGenerator());
+        defaultFlowGenerators.add(switchFlowFactory.getUnicastVerificationVxlanFlowGenerator());
```

Comment:
I'm not sure that we must add generator if we (as developers) know that flow has no meters. But this case is dissoluble.

Reply:
Reworked.